algorithms/src/Kokkos_Random.hpp

```diff
@@ -751,7 +751,9 @@ class Random_XorShift64 {
   float frand() { return urand64() / static_cast<float>(MAX_URAND64); }
 
   KOKKOS_INLINE_FUNCTION
-  float frand(const float& range) { return range * urand64() / MAX_URAND64; }
+  float frand(const float& range) {
+    return range * urand64() / static_cast<float>(MAX_URAND64);
```

Comment:
is there any use in doing `range * static_cast<float>(urand64()) / static_cast<float>(MAX_URAND64)` instead? Also I am wondering whether we should do the division in double precision and then scale down to float.

Reply:
The return value of `urand64()` is implicitly cast anyway so it doesn't make a difference to also `static_cast` it. 
I don't see an advantage in doing the division in double precision. What were your thoughts?

Reply:
essentially my thought was doing the division in 64 bit (or at least 48bit or whatever the mantisse is). I am wondering what is the number of distinct values coming from frand vs drand vs urand32 vs urand64. I am not sure that I thought that fully through.

Reply:
i.e. would range*(double(urand64())/double(MAX_URAND64)) have more distinct answers than (range * float(urand64())) / float(MAX_URAND64) ?

Reply:
maybe I should try this on a GPU, reinterprete_cast the answer into uint32_t and make a 4billion wide histogram with a couple trillion pulls ...